lib/membrane_rtc_engine/endpoints/hls_endpoint.ex

```diff
@@ -277,6 +277,111 @@ if Enum.all?(
       {{:ok, spec: spec}, state}
     end
 
+    @impl true
+    def handle_pad_added(Pad.ref(:input, track_id) = pad, _ctx, %{hls_mode: :muxed_av} = state) do
+      link_builder = link_bin_input(pad)
+      track = Map.get(state.tracks, track_id)
+      directory = Path.join(state.output_directory, track.stream_id)
+      track_type = if track.encoding == :H264, do: :video, else: :audio
+
+      link = [
+        link_builder
+        |> via_in(Pad.ref(:input, track_type))
+        |> to({:track_sync, track.stream_id})
+      ]
```

Comment:
This will synchronize only two tracks; one audio and one video. We should mention it in the docs. Consider scenario when someone is sharing its screen. Video from screen share won't be synchronized with audio from microphone and video from camera